packages/wallet/src/Wallets/BaseWallet.ts

```diff
@@ -821,6 +843,41 @@ export class BaseWallet implements ObservableWallet {
     return firstValueFrom(this.addresses$);
   }
 
+  async getNextUnusedAddress(): Promise<WalletAddress | null> {
```

Comment:
Based on 
> As for the second change request, I tought we commit to only having one payment crendetial (index 0) asociated with other stake keys, only stake key allowed to have more than one payment credential is the first one (the discovery assumes this)

There would be no use in the future for more than 1 unused address, right? https://github.com/input-output-hk/cardano-js-sdk/pull/1405#discussion_r1701347073
```suggestion
  async getNextUnusedAddresses(): Promise<WalletAddress[]> {
```

Reply:
done here https://github.com/input-output-hk/cardano-js-sdk/pull/1405/commits/0b348d9124411f211994c69acc698a11c1468782